src/header-bidding/prebid/appnexus.spec.ts

```diff
@@ -0,0 +1,209 @@
+import { createAdSize } from 'core/ad-sizes';
+import {
+	isInAuOrNz as isInAuOrNz_,
+	isInRow as isInRow_,
+	isInUk as isInUk_,
+	isInUsa as isInUsa_,
+} from 'utils/geo-utils';
+import {
+	containsBillboardNotLeaderboard as containsBillboardNotLeaderboard_,
+	containsDmpu as containsDmpu_,
+	containsLeaderboard as containsLeaderboard_,
+	containsLeaderboardOrBillboard as containsLeaderboardOrBillboard_,
+	containsMobileSticky as containsMobileSticky_,
+	containsMpu as containsMpu_,
+	containsMpuOrDmpu as containsMpuOrDmpu_,
+	getBreakpointKey as getBreakpointKey_,
+} from '../utils';
+import { getAppNexusDirectPlacementId } from './appnexus';
+
+jest.mock('../utils');
+const containsBillboardNotLeaderboard =
+	containsBillboardNotLeaderboard_ as jest.Mock;
+const containsDmpu = containsDmpu_ as jest.Mock;
+const containsLeaderboard = containsLeaderboard_ as jest.Mock;
+const containsLeaderboardOrBillboard =
+	containsLeaderboardOrBillboard_ as jest.Mock;
+const containsMobileSticky = containsMobileSticky_ as jest.Mock;
+const containsMpu = containsMpu_ as jest.Mock;
+const containsMpuOrDmpu = containsMpuOrDmpu_ as jest.Mock;
+const getBreakpointKey = getBreakpointKey_ as jest.Mock;
+
+jest.mock('utils/geo-utils');
+const isInAuOrNz = isInAuOrNz_ as jest.Mock;
+const isInRow = isInRow_ as jest.Mock;
+const isInUk = isInUk_ as jest.Mock;
+const isInUsa = isInUsa_ as jest.Mock;
+
+jest.mock('experiments/ab', () => ({
+	isInVariantSynchronous: jest.fn(),
+}));
+
+jest.mock('lib/cookies', () => ({
+	getCookie: jest.fn(),
+}));
+
+const resetConfig = () => {
+	window.guardian.ophan = {
+		pageViewId: 'pvid',
+		viewId: 'v_id',
+		record: () => {
+			// do nothing;
+		},
+		setEventEmitter: null,
+		trackComponentAttention: null,
+	};
+	window.guardian.config.switches = {
+		prebidAppnexus: true,
+		prebidAppnexusInvcode: true,
+		prebidOpenx: true,
+		prebidImproveDigital: true,
+		prebidIndexExchange: true,
+		prebidSonobi: true,
+		prebidTrustx: true,
+		prebidXaxis: true,
+		prebidAdYouLike: true,
+		prebidTriplelift: true,
+		prebidCriteo: true,
+	};
+	window.guardian.config.page.contentType = 'Article';
+	window.guardian.config.page.section = 'Magic';
+	window.guardian.config.page.isDev = false;
+};
+
+describe('getAppNexusDirectPlacementId', () => {
+	afterEach(() => {
+		jest.resetAllMocks();
+		resetConfig();
+	});
+
+	test('should return correct placementID for any existing biding sizes in Australia or New Zealand', () => {
```

Comment:
biding -> bidding

Reply:
Done in 12648f9372f9fb09dbe7a793a47d6c6bdbe748aa